### PR TITLE
tests/pushbits: ignore ESRCH errors from pthread_cancel

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
Strictly speaking, a thread ID should still valid until pthread_join() (ignoring a few other quibbles).  This was fixed in FreeBSD:
https://lists.freebsd.org/pipermail/svn-src-head/2020-February/133736.html

However, many platforms still return ESRCH in that case, so for compatibility we ignore them.

File: tests/pushbits/main.c [hunk 1]
```diff
@@ -77,8 +77,17 @@ echo_stdin_stdout(void)
...
 	sleep(1);
 	if ((rc = pthread_cancel(thread)) != 0) {
-		warn0("pthread_cancel: %s", strerror(rc));
-		goto err0;
+		/*
+		 * These tests ignore an ESRCH error returned by
+		 * pthread_cancel.  According to the POSIX standard, a
+		 * thread ID should still be valid once it finishes if
+		 * pthread_join() has not yet been called.  However,
+		 * many platforms return ESRCH in the situation.
```
Comment (cperciva) [inline tests/pushbits/main.c]: in *this* situation?

File: tests/pushbits/main.c [hunk 2]
```diff
@@ -77,8 +77,17 @@ echo_stdin_stdout(void)
 	/* Wait a short while, then cancel the thread. */
 	sleep(1);
 	if ((rc = pthread_cancel(thread)) != 0) {
-		warn0("pthread_cancel: %s", strerror(rc));
-		goto err0;
+		/*
+		 * These tests ignore an ESRCH error returned by
+		 * pthread_cancel.  According to the POSIX standard, a
+		 * thread ID should still be valid once it finishes if
```
Comment (cperciva) [inline tests/pushbits/main.c]: "once it finishes" is a bit imprecise (does the thread ID finish?) so I'd go with "after pthread_exit has been invoked by the thread".

**Comment by gperciva** [comment]
Updated with the two changes to the comment.

**Pull request merged**